examples/pylab_examples/newscalarformatter_demo.py

```diff
@@ -1,88 +1,92 @@
-#!/usr/bin/env python
-# Demonstrating the improvements and options of the proposed new ScalarFormatter
-from pylab import *
-from matplotlib.ticker import OldScalarFormatter
+# from pylab import *
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib.ticker import OldScalarFormatter, ScalarFormatter
 
-x = frange(0, 1, .01)
-f = figure(figsize=(6, 6))
-f.text(0.5, 0.975, 'The old formatter', horizontalalignment='center', verticalalignment='top')
-subplot(221)
-plot(x*1e5 + 1e10, x*1e-10 + 1e-5)
-gca().xaxis.set_major_formatter(OldScalarFormatter())
-gca().yaxis.set_major_formatter(OldScalarFormatter())
-subplot(222)
-plot(x*1e5, x*1e-4)
-gca().xaxis.set_major_formatter(OldScalarFormatter())
-gca().yaxis.set_major_formatter(OldScalarFormatter())
-subplot(223)
-plot(-x*1e5 - 1e10, -x*1e-5 - 1e-10)
-gca().xaxis.set_major_formatter(OldScalarFormatter())
-gca().yaxis.set_major_formatter(OldScalarFormatter())
-subplot(224)
-plot(-x*1e5, -x*1e-4)
-gca().xaxis.set_major_formatter(OldScalarFormatter())
-gca().yaxis.set_major_formatter(OldScalarFormatter())
+x = np.arange(0, 1, .01)
+f = plt.figure(figsize=(6, 6))
+f.text(0.5, 0.975, 'The old formatter',
+       horizontalalignment='center', verticalalignment='top')
+plt.subplot(221)
+plt.plot(x * 1e5 + 1e10, x * 1e-10 + 1e-5)
+plt.gca().xaxis.set_major_formatter(OldScalarFormatter())
```

Comment:
Can you capture this as `ax = plt.gca()` and then use that for the rest of the example?